pytket/pytket/qir/qir.py

```diff
@@ -86,37 +161,63 @@ def circuit_to_qir_str(circ: Circuit, root: str) -> str:
         )
     ):
         raise QIRUnsupportedError("Complex classical gates not supported.")
-    module = SimpleModule(root, num_qubits=circ.n_qubits, num_results=len(circ.bits))
-    qis = BasicQisBuilder(module.builder)
-
     for command in circ:
         op = command.op
-        qubits = _to_qis_qubit(command.qubits, module)
         optype, params = _get_optype_and_params(op)
-        if optype == OpType.Measure:
+        if isinstance(module, ExtendedModule):
+            mod = module.module
+            qubits = _to_qis_qubits(command.qubits, mod)
+            results = _to_qis_results(command.bits, mod)
+            try:
+                gate = module.gateset.tk_to_gateset(optype)
+            except KeyError:
+                raise KeyError(
+                    "Gate not defined in {:} gate set.".format(module.gateset.name)
+                )
+            get_gate = getattr(module, gate)
+            if params:
+                mod.builder.call(get_gate, [*params, *qubits])
```

Comment:
if you return an empty list from `_get_optype_and_params` if there are no params rather than `None`, this case will cover the `else` case as well